test/forge/HalmosTest.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import "../../lib/forge-std/src/Test.sol";
+import {SymTest} from "../../lib/halmos-cheatcodes/src/SymTest.sol";
+
+import {IMorpho} from "../../src/interfaces/IMorpho.sol";
+import {IrmMock} from "../../src/mocks/IrmMock.sol";
+import {ERC20Mock} from "../../src/mocks/ERC20Mock.sol";
+import {OracleMock} from "../../src/mocks/OracleMock.sol";
+
+import "../../src/Morpho.sol";
+import "../../src/libraries/ConstantsLib.sol";
+import {MorphoLib} from "../../src/libraries/periphery/MorphoLib.sol";
+
+/// @custom:halmos --symbolic-storage --solver-timeout-assertion 0
+contract HalmosTest is SymTest, Test {
+    using MorphoLib for IMorpho;
+    using MarketParamsLib for MarketParams;
+
+    uint256 internal constant BLOCK_TIME = 1;
+
+    address internal OWNER;
+    address internal FEE_RECIPIENT;
+
+    IMorpho internal morpho;
+    ERC20Mock internal loanToken;
+    ERC20Mock internal collateralToken;
+    OracleMock internal oracle;
+    IrmMock internal irm;
+
+    MarketParams internal marketParams;
+    Id internal id;
+
+    function setUp() public virtual {
+        OWNER = address(0x10);
+        FEE_RECIPIENT = address(0x11);
+
+        morpho = IMorpho(address(new Morpho(OWNER)));
+
+        loanToken = new ERC20Mock();
+        collateralToken = new ERC20Mock();
+        oracle = new OracleMock();
+        oracle.setPrice(ORACLE_PRICE_SCALE);
+        irm = new IrmMock();
+
+        vm.startPrank(OWNER);
+        morpho.enableIrm(address(0));
+        morpho.enableIrm(address(irm));
+        morpho.enableLltv(0);
+        morpho.setFeeRecipient(FEE_RECIPIENT);
+        vm.stopPrank();
+
+        uint256 lltv = svm.createUint256("lltv");
+        marketParams = MarketParams(address(loanToken), address(collateralToken), address(oracle), address(irm), lltv);
+        id = marketParams.id();
+
+        vm.assume(block.timestamp != 0);
```

Comment:
Yes it has to be symbolic otherwise there would be no valid path of execution. This is because if it was not symbolic it would be at the default `0` value, and this would make it trivial with this line.
If it's zero then you would have a created market with `lastUpdate = 0`, so the rule `check_lastUpdateNonZero` would fail